BlueBrain/Search
# [BBS-292] Release v0.1.0

Fixes [BBS-292](https://bbpteam.epfl.ch/project/issues/browse/BBS-292).

## Description

1. **Remove GitHub Actions workflow to automatically upload to PyPI.** This was done after discussing with @jankrepl about pros and cons of this choice. See https://github.com/BlueBrain/atlas_alignment/pull/19 for the rationale.
2. **DVC Dockerfiles use versions specified in`requirements.txt`**. This change improves the reproducibility of results with DVC. Before this, we were just using the dependencies specified in `setup.py` which means that there was no guarantee on the version of each dependency used to run the DVC pipelines and results could have slightly changed.
3. **Upgrade `prodigy`** version to fix [this error](https://support.prodi.gy/t/importerror-cannot-import-name-schema-from-pydantic/3948). 
4. **Run `dvc repro`** (and `dvc push` to update the remote storage).
5. **DVC-untrack BSV** model. This reduces the `.dvc/cache` to only `4.4 GiB`.


## How to test?
1. **Get Prodigy wheel**. 
```cp /raid/prodigy_downloads/1.10.7/prodigy-1.10.7-cp36.cp37.cp38-cp36m.cp37m.cp38-linux_x86_64.whl path/to/Search```
1. **Get images for running the `ner` and `sentemb` pipelines.** You have two options.
     1. **Option1** — You can use the images I already built on DGX1: `bbs_dvc_ner:v0.1.0` and `bbs_dvc_sentemb:v0.1.0`. In the following I assume you use this option—otherwise replace the image names in the following commands! 
     1. **Option 2** — You can build the images yourself: 
         - `docker build -f data_and_models/pipelines/ner/Dockerfile  -t <ner_image_name> .` 
         - `docker build -f data_and_models/pipelines/sentence_embedding/Dockerfile  -t <sentemb_image_name> .` 
1. **Reproduce the `ner` pipeline.**
    1. Get all dvc files `dvc pull`
    2. Run container`docker run --rm -it <your_options> --name <ner_container_name> bbs_dvc_ner:v0.1.0`
    2. Move to directory `cd data_and_models/pipelines/ner/`
    3. Reproduce pipeline `dvc repro` (or `dvc repro -f` if you have some time and want to be 100% sure)
    4. Check that nothing changed: `git diff` should be empty
1. **Reproduce the `sentemb` pipeline.**
    1. Get all dvc files `dvc pull` (if you run the `ner` pipeline already, then this step can be skipped)
    2. Run container`docker run --rm -it <your_options> --name <sentemb_container_name> bbs_dvc_sentemb:v0.1.0`
    2. Move to directory `cd data_and_models/pipelines/sentence_embedding/`
    3. Reproduce pipeline `dvc repro` (or `dvc repro -f` if you have some time and want to be 100% sure)
    4. Check that nothing changed: `git diff` should be empty


## Checklist

  (if it is not the case, please create an issue first).
  (if needed)
